src/object/endpoint.c

```diff
@@ -226,7 +226,9 @@ void receiveIPC(tcb_t *thread, cap_t cap, bool_t isBlocking)
             if (do_call ||
                 seL4_Fault_get_seL4_FaultType(sender->tcbFault) != seL4_Fault_NullFault) {
                 if ((canGrant || canGrantReply) && replyPtr != NULL) {
-                    reply_push(sender, thread, replyPtr, sender->tcbSchedContext != NULL);
+                    bool_t canDonate = sender->tcbSchedContext != NULL
```

Comment:
Nit: can the commit message be updated to 'prevent recursion on timeout faults'?

Reply:
Sure!

Reply:
Done!